Util.py

```diff
@@ -2183,25 +2183,35 @@ def unicode_to_str(s):
   return s
 
 
-def deepcopy(x):
+def deepcopy(x, stop_types=None):
   """
   Simpler variant of copy.deepcopy().
   Should handle some edge cases as well, like copying module references.
 
   :param T x: an arbitrary object
+  :param list[type]|None stop_types: objects of these types will not be deep-copied, only the reference is passed
   :rtype: T
   """
   # See also class Pickler from TaskSystem.
   # Or: https://mail.python.org/pipermail/python-ideas/2013-July/021959.html
   from TaskSystem import Pickler, Unpickler
 
+  persistent_memo = {}
+
+  def persistent_id(obj):
+    if any([isinstance(obj, stop_type) for stop_type in (stop_types or [])]):
```

Comment:
This is way too complicated and hard to read.
Just do
```suggestion
    if stop_types and isinstance(obj, stop_types):
```

Reply:
It was not clear to me that isinstance takes lists, I will update this

Reply:
It takes tuples. Maybe we should enforce the type to be a tuple (then you must fix the usage), or convert to tuple here.
```suggestion
    if stop_types and isinstance(obj, tuple(stop_types)):
```

Reply:
`isinstance` does not accept lists, but tuples work